src/m365/pa/commands/app/app-permission-remove.spec.ts

```diff
@@ -0,0 +1,397 @@
+import * as assert from 'assert';
+import * as sinon from 'sinon';
+import { telemetry } from '../../../../telemetry';
+import auth from '../../../../Auth';
+import { Cli } from '../../../../cli/Cli';
+import { CommandInfo } from '../../../../cli/CommandInfo';
+import { Logger } from '../../../../cli/Logger';
+import Command, { CommandError } from '../../../../Command';
+import request from '../../../../request';
+import { pid } from '../../../../utils/pid';
+import { session } from '../../../../utils/session';
+import { sinonUtil } from '../../../../utils/sinonUtil';
+import commands from '../../commands';
+import { aadUser } from '../../../../utils/aadUser';
+import { aadGroup } from '../../../../utils/aadGroup';
+import { accessToken } from '../../../../utils/accessToken';
+const command: Command = require('./app-permission-remove');
+
+describe(commands.APP_PERMISSION_REMOVE, () => {
+  let cli: Cli;
+  let log: string[];
+  let logger: Logger;
+  let commandInfo: CommandInfo;
+
+  const validEnvironmentName = 'Default-6a2903af-9c03-4c02-a50b-e7419599925b';
+  const validAppName = '784670e6-199a-4993-ae13-4b6747a0cd5d';
+  const validUserId = 'd2481133-e3ed-4add-836d-6e200969dd03';
+  const validUserName = 'john.doe@contoso.com';
+  const validGroupId = 'c6c4b4e0-cd72-4d64-8ec2-cfbd0388ec16';
+  const validGroupName = 'CLI Group';
+  const appPermissionRemoveResponse = { put: [] };
+  const groupResponse = {
+    "id": validGroupId,
+    "deletedDateTime": null,
+    "classification": null,
+    "createdDateTime": "2017-11-29T03:27:05Z",
+    "description": "This is the Contoso Finance Group. Please come here and check out the latest news, posts, files, and more.",
+    "displayName": "Finance",
+    "groupTypes": [
+      "Unified"
+    ],
+    "mail": "finance@contoso.onmicrosoft.com",
+    "mailEnabled": true,
+    "mailNickname": "finance",
+    "onPremisesLastSyncDateTime": null,
+    "onPremisesProvisioningErrors": [],
+    "onPremisesSecurityIdentifier": null,
+    "onPremisesSyncEnabled": null,
+    "preferredDataLocation": null,
+    "proxyAddresses": [
+      "SMTP:finance@contoso.onmicrosoft.com"
+    ],
+    "renewedDateTime": "2017-11-29T03:27:05Z",
+    "securityEnabled": false,
+    "visibility": "Public"
+  };
+  const tenantId = '174290ec-373f-4d4c-89ea-9801dad0acd9';
+
+  before(() => {
+    cli = Cli.getInstance();
+    sinon.stub(auth, 'restoreAuth').resolves();
+    sinon.stub(telemetry, 'trackEvent').returns();
+    sinon.stub(pid, 'getProcessName').returns('');
+    sinon.stub(session, 'getId').returns('');
+    auth.service.connected = true;
+    if (!auth.service.accessTokens[auth.defaultResource]) {
```

Comment:
We should always clear this value when the testing has been done, so let's remove the if and clear it after running the tests.